### PR TITLE
Feature/spring boot 3 fix bugs caused by supporting azure ad proxy

### PR DESCRIPTION
**Pull request by chenrujun**
Fix these bugs:
 - #31190
 - #31191
 - #31482 
 - #31593
 - #31593

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @saragluna , @moarychan , please help to review this PR.

**Comment by chenrujun** [comment]
The failure of [java - code - quality reports - ci](https://github.com/Azure/azure-sdk-for-java/pull/31684/checks?check_run_id=9060036923) can be ignored because it not caused by this PR.

The failure of [java - spring - tests](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1938719&view=logs&j=32044c8a-cbe9-5d25-6615-786933ab9334&t=8fea6837-1f61-5208-5da0-48288d222902) (**EventHubsBinderConsumeErrorIT.integrationTest**) can be ignored because it's not caused by this PR. And @moarychan is fixing it.

File: .vscode/cspell.json [hunk 1]
```diff
@@ -248,6 +248,13 @@
         "compilerArguments",
         "annotationProcessors"
       ]
+    },
+    {
+      "filename": "sdk/spring/CHANGELOG.md",
+      "words": [
+        "AADB",
```
Comment (saragluna) [inline .vscode/cspell.json:255]: what is AADB?

Comment (chenrujun) [reply .vscode/cspell.json:255]: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/spring/CHANGELOG.md#breaking-changes-3

image

**Review by moarychan — APPROVED** [review]
LGTM, thanks Rujun!

**Pull request merged**